src/test/unit/test.js

```diff
@@ -491,6 +491,51 @@ describe('Client', function() {
     })
   })
```

Comment:
> One issue with functional tests instead of unit tests in this case is we couldn't ensure that the outgoing request has the same payload each time — we'd only be testing that the server interpreted it correctly, as opposed to testing whether or not we sent the right payload.

For that you can add a test case for `generateBucketPolicy()` to check if it returns the expected output. Unit tests for our internal functions will be good to have, see if you can add it. But we can do away with mock tests (involving http calls).

But functional tests are more important i.e testing things against actual server as it will be more of a real world test.
But having both mock tests and functional tests will be too much, so we prefer functional tests over mock tests.

does that sound right?